docs: Add plugins showcase (#1275)
withastro/starlight: docs/src/content/docs/showcase.mdx

See all the [public project repos using Starlight on GitHub](https://github.com/withastro/starlight/network/dependents).

## Plugins

Starlight [plugins](/reference/plugins/) can customize Starlight configuration, UI, and behavior, while also being easy to share and reuse.
This section lists the official plugins supported by the Starlight team and also community plugins that are maintained by Starlight users.

> ### Algolia DocSearch

Reviewer: I don't know if we are planning to support more official plugins in the future but I was thinking that maybe we could have an official plugins section and list the algolia plugin under that section.

I think this will help level the importance of official and community plugins which will encourage more people to share their creations.

Author: Yeah, I think you're totally right. I updated the PR to add an "Official plugins" section and if we want to go back to a structure closer to the [suggested one](https://github.com/withastro/starlight/pull/942/commits/d3ef3114e88b6b59d91d1c5e94435ec43b7213cc), we can easily revert or tweak it.

See all the [public project repos using Starlight on GitHub](https://github.com/withastro/starlight/network/dependents).

## Plugins

> Starlight [plugins](/reference/plugins/) can customize Starlight configuration, UI, and behavior, while also being easy to share and reuse.

Reviewer 2: Tiny nit suggestion to avoid the repetition of “Starlight”:

```suggestion
[Plugins](/reference/plugins/) can customize Starlight configuration, UI, and behavior, while also being easy to share and reuse.
```

> This section lists the official plugins supported by the Starlight team and also community plugins that are maintained by Starlight users.

Reviewer 2: Bit subjective this one, but I wanted to avoid “this section lists…” because it seemed a bit pedestrian and obvious, but I like how the sentence explains the maintenance difference between official and community plugins. What do you think of this?

```suggestion
Extend your site with official plugins supported by the Starlight team and community plugins maintained by Starlight users.
```

Author: Great suggestion, this is indeed way better while keeping the maintenance difference :thumbsup:
